File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-5.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n = len(nums)
        start = 0
        end = n-1

        while start < end :
            mid = int(start + (end-start)/2)
            if nums[mid] > nums[mid+1] :
                return nums[mid+1]
            elif nums[mid] > nums[end] :
                start = mid+1
            else :
                end = mid

        
        return nums[0]

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        int max = 0;

        for(int num : nums) {
            set.add(num);
        }

        for(int num : set) {
            if(!set.contains(num-1)){
                int longest = 1;
                while(set.contains(num+longest)){
                    longest++;
                }
                max = Math.max(max, longest);
            }
        }

        return max;
    }
}
